src/MongoDB.EntityFrameworkCore/Infrastructure/MongoModelRuntimeInitializer.cs

```diff
@@ -0,0 +1,66 @@
+﻿/* Copyright 2023-present MongoDB Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Diagnostics;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.Metadata;
+using MongoDB.Bson;
+using MongoDB.Bson.Serialization;
+using MongoDB.Bson.Serialization.Conventions;
+
+namespace MongoDB.EntityFrameworkCore.Infrastructure;
+
+/// <summary>
+/// Initializes a <see cref="IModel" /> with the runtime dependencies.
+/// </summary>
+/// <param name="dependencies"></param>
+public class MongoModelRuntimeInitializer(ModelRuntimeInitializerDependencies dependencies)
+    : ModelRuntimeInitializer(dependencies)
+{
+    private static readonly Dictionary<Type, IDiscriminatorConvention>? DiscriminatorConventionDictionary =
```

Comment:
This doesn't seem to be used right now. Is it needed?

Reply:
Oops! This PR took the code from the upcoming type disciminator PR but I accidentally left that in. Now removed. Thanks!